hermes/wasm/crates/byte-array-literals/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//! Converts `str` literals into byte arrays.
+
+use proc_macro::{Delimiter, Group, Literal, Punct, Spacing, TokenStream, TokenTree};
+
+/// Expand a `str` literal into a byte array.
+#[proc_macro]
+pub fn str(input: TokenStream) -> TokenStream {
+    let rv = convert_str(input);
+
+    vec![TokenTree::Group(Group::new(
+        Delimiter::Bracket,
+        rv.into_iter().collect(),
+    ))]
+    .into_iter()
+    .collect()
+}
+
+/// The same as `str` but appends a `'\n'`.
+#[proc_macro]
+pub fn str_nl(input: TokenStream) -> TokenStream {
```

Comment:
It's originally from [this source,](https://github.com/bytecodealliance/wasmtime/blob/main/crates/wasi-preview1-component-adapter/byte-array-literals/src/lib.rs). It might be preferable to keep it as is.

Reply:
yea @apskhem right, we don't want to change the source code of these crates, unless it's for some small changes to make `linting` happy